### PR TITLE
*: use gogo, update proto

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut  @shenli @BusyJay @disksing @huachaohuang

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@qiuyesuifeng

File: proto/metapb.proto [hunk 1]
```diff
@@ -8,29 +8,29 @@ option (gogoproto.sizer_all) = true;
...
-    optional uint64 conf_ver	= 1;
+    optional uint64 conf_ver	= 1 [(gogoproto.nullable) = false];
     // Region version, auto increment when split or merge
-    optional uint64 version     = 2;
+    optional uint64 version     = 2 [(gogoproto.nullable) = false];
 }
 
 message Region {
-    optional uint64 id                  = 1;
+    optional uint64 id                  = 1 [(gogoproto.nullable) = false];
```
Comment (qiuyesuifeng) [inline proto/metapb.proto]: Should we add gogoproto tag for `start_key` and `end_key `?

Comment (siddontang) [inline proto/metapb.proto]: no need, gogo will throw warning if we set this flag for bytes.

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**